man/man3/Sequence.3
.TH "Sequence" 3 "Mon Feb 13 2023" "My Project" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Sequence
.SH SYNOPSIS
.br
.PP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBSequence\fP (int=100, int=25, string='red')"
.br
.RI "konstruktor domyślny "
.ti -1c
.RI "\fB~Sequence\fP ()"
.br
.RI "destruktor "
.ti -1c
.RI "\fBSequence\fP (const \fBSequence\fP &fSequence)"
.br
.RI "konstruktor kopiujący "
.ti -1c
.RI "int \fBgetWidth\fP ()"
.br
.RI "funkcja zwracająca szerokość obiektu "
.ti -1c
.RI "void \fBsetWidth\fP (int)"
.br
.RI "funkcja pozwalająca ustawić szerokość obiektu "
.ti -1c
.RI "int \fBgetHeight\fP ()"
.br
.RI "funkcja zwracająca wysokość obiektu "
.ti -1c
.RI "void \fBsetHeight\fP (int)"
.br
.RI "funkcja pozwalająca ustawić wysokość obiektu "
.ti -1c
.RI "string \fBgetColor\fP ()"
.br
.RI "funkcja zwracająca kolor obiektu "
.ti -1c
.RI "void \fBsetColor\fP (string)"
.br
.RI "funkcja pozwalająca ustawić kolor obiektu "
.ti -1c
.RI "double \fBcalcSimpleIntegral\fP ()"
.br
.RI "funkcja obliczająca silnię danej liczby "
.ti -1c
.RI "double \fBcalcComplexIntegral\fP ()"
.br
.RI "funkcja obliczająca skomplikowaną całkę "
.ti -1c
.RI "double \fBheightOfEquTriangle\fP ()"
.br
.RI "funkcja obliczająca wysokość trójkąta równobocznego na podstawie długości ramienia "
.ti -1c
.RI "double \fBsin\fP (double)"
.br
.RI "funkcja obliczająca sinus danego kąta "
.ti -1c
.RI "double \fBcos\fP (double)"
.br
.RI "funkcja obliczająca cosinus danego kąta "
.in -1c
.SH "Member Function Documentation"
.PP 
.SS "Sequence::calcComplexIntegral ()"

.PP
funkcja obliczająca skomplikowaną całkę Funkcja bazuje w oparciu o wzory 
.br
 $ \int\frac{dx}{x\sqrt{x^2-1}} = -cosec^{-1}x+C $ 
.br
 $ \int sin^2 x dx = \int\frac{1-cos2x}{2} dx $
.SS "Sequence::calcSimpleIntegral ()"

.PP
funkcja obliczająca silnię danej liczby Funkcja bazuje w oprarciu o wzór $ \int\frac{1}{\sqrt{x^2 - a^2}}dx = ln|x+\sqrt{x^2-a^2}|+C $
.SS "Sequence::cos (double x)"

.PP
funkcja obliczająca cosinus danego kąta funkcja oblicza cosinus danego kąta na podstawie kąta podanego w stopniach kątowych 
.PP
\fBParameters\fP
.RS 4
\fIx\fP - kąt w stopniach kątowych
.RE
.PP
RadianyWartość cosinusStopnie $\frac{Π}{2}$090 $\frac{Π}{3}$$\frac{1}{2}$ 60 $\frac{Π}{4}$$\frac{\sqrt{2}}{2}$45 $\frac{Π}{6}$$\frac{\sqrt{3}}{2}$30 000 
.PP
Jeśli wykres nie działa należy w doxygenie w image path dodać ścieżke img
.PP
Wykres cosinusa      
.SS "Sequence::heightOfEquTriangle ()"

.PP
funkcja obliczająca wysokość trójkąta równobocznego na podstawie długości ramienia wysokość trójkąta o boku a to $h=\frac{a\times\sqrt3}{2} $
.SS "Sequence::sin (double x)"

.PP
funkcja obliczająca sinus danego kąta funkcja oblicza sinus danego kąta na podstawie kąta podanego w stopniach kątowych 
.PP
\fBParameters\fP
.RS 4
\fIx\fP - kąt w stopniach kątowych
.RE
.PP
RadianyWartość sinStopnie $\frac{Π}{2}$190 $\frac{Π}{3}$$\frac{\sqrt{3}}{2}$ 60 $\frac{Π}{4}$$\frac{\sqrt{2}}{2}$45 $\frac{Π}{6}$$\frac{1}{2}$30 000 
.PP
Jeśli wykres nie działa należy w doxygenie w image path dodać ścieżke img
.PP
Wykres Sinusa      

.SH "Author"
.PP 
Generated automatically by Doxygen for My Project from the source code\&.


sequence.cpp
#include <iostream>
#include "sequence.h"
#include <cmath>
using namespace std;

Sequence::Sequence(int w, int h, string c)
{
    width w;
    height = h;
    color = c;
}

Sequence::~Sequence()
{

}

Sequence::Sequence(const Sequence & fSequence)
{
    width = new int;
    height = new int;
    color = new string;
    *width = *fSequence.width;
    *height = *fSequence.height;
    *color = *fSequence.color;
}

Sequence::getWidth()
{
    return width;
}
Sequence::setWidth(int w)
{
    width = w;
}
Sequence::getHeight()
{
    return height;
}
Sequence::setHeight(int h)
{
    height = h;
}
Sequence::getColor()
{
    return color;
}
Sequence::setColor(string c)
{
    color = c;
}
Sequence::calcSimpleIntegral()
{
    /**
        Funkcja bazuje w oprarciu o wzór
       \f$ \int\frac{1}{\sqrt{x^2 - a^2}}dx = ln|x+\sqrt{x^2-a^2}|+C \f$

    */
}
Sequence::calcComplexIntegral()
{
    /**
        Funkcja bazuje w oparciu o wzory \n
        \f$ \int\frac{dx}{x\sqrt{x^2-1}} = -cosec^{-1}x+C \f$ \n
        \f$ \int sin^2 x dx = \int\frac{1-cos2x}{2} dx \f$
    */

}
Sequence::heightOfEquTriangle(int a)
{
    //! wysokość trójkąta o boku a to \f$h=\frac{a\times\sqrt3}{2} \f$
    float h = (a*sqrt(3))/2;
    return h;
}
Sequence::sin(double x) {

    /**
        funkcja oblicza sinus danego kąta na podstawie
        kąta podanego w stopniach kątowych
        \param[in] x - kąt w stopniach kątowych

        <table>
            <tr><th>Radiany<th>Wartość sin</th><th>Stopnie</th></tr>
            <tr><td>\f$\frac{Π}{2}\f$</td><td>1</td><td>90</td></tr>
            <tr><td>\f$\frac{Π}{3}\f$</td><td> \f$\frac{\sqrt{3}}{2}\f$ </td><td>60</td></tr>
            <tr><td>\f$\frac{Π}{4}\f$</td><td>\f$\frac{\sqrt{2}}{2}\f$</td><td>45</td></tr>
            <tr><td>\f$\frac{Π}{6}\f$</td><td>\f$\frac{1}{2}\f$</td><td>30</td></tr>
            <tr><td>0</td><td>0</td><td>0</td></tr>
        </table>

        Jeśli wykres nie działa należy w doxygenie w image path dodać ścieżke img

        | Wykres Sinusa |
        | :---- |
        | \image html sinus.bmp |
    */
}
Sequence::cos(double x)
{
    /**
        funkcja oblicza cosinus danego kąta na podstawie
        kąta podanego w stopniach kątowych
        \param[in] x - kąt w stopniach kątowych

        <table>
            <tr><th>Radiany<th>Wartość cosinus</th><th>Stopnie</th></tr>
            <tr><td>\f$\frac{Π}{2}\f$</td><td>0</td><td>90</td></tr>
            <tr><td>\f$\frac{Π}{3}\f$</td><td> \f$\frac{1}{2}\f$ </td><td>60</td></tr>
            <tr><td>\f$\frac{Π}{4}\f$</td><td>\f$\frac{\sqrt{2}}{2}\f$</td><td>45</td></tr>
            <tr><td>\f$\frac{Π}{6}\f$</td><td>\f$\frac{\sqrt{3}}{2}\f$</td><td>30</td></tr>
            <tr><td>0</td><td>0</td><td>0</td></tr>
        </table>

        Jeśli wykres nie działa należy w doxygenie w image path dodać ścieżke img

        | Wykres cosinusa |
        | :---- |
        | \image html cosinus.jpg |
    */


}
